### DISCUSSION TITLE
indexColumns Date format

### DISCUSSION THREAD

**Discussion by michaeljcoyne** [area17/twill#1318] [General]
Hi is there any way to format the indexColumns to be  a user firendly date format?

Eg. created_at

Thanks

**Comment by pboivin** [comment]
Hi @michaeljcoyne,

There a different ways to approach this but these days, I tend to use attribute accessors. Quick example:

In your controller:

```
    protected $indexColumns = [
        'title' => [
            'title' => 'Title',
            'field' => 'title',
        ],
        'created_at' => [
            'title' => 'Created',
            'field' => 'formatted_created_at',
        ],
    ];
```

In your model:

```
    public function getFormattedCreatedAtAttribute()
    {
        return $this->created_at->format('Y-m-d');
    }
```

The advantage is to be able to reuse this anywhere in your app: `$item->formatted_created_at`.

**Reply by michaeljcoyne** [reply]
Aha perfect, thanks.

**Reply by michaeljcoyne** [reply]
@pboivin This doesn't work for when sortable as the key sorkKey param is now using formatted_created_at. Is there a way to overide this key in the model?

**Reply by michaeljcoyne** [reply]
And the Laravel $casts does not seem, to work.
